app/bundles/CampaignBundle/Entity/LeadEventLogRepository.php

```diff
@@ -661,4 +661,58 @@ public function isLastFailed(int $leadId, int $eventId): bool
 
         return false;
     }
+
+    /**
+     * @return array{
+     *     total_executions: int,
+     *     pending_executions: int,
+     *     negative_path_count: int,
+     *     positive_path_count: int,
+     *     first_execution_date: string|null,
+     *     last_execution_date: string|null
+     * }
+     */
+    public function getEventLogStats(int $eventId): array
+    {
+        $qb = $this->getReplicaConnection()->createQueryBuilder();
+        $qb->select([
+            'COUNT(log.id) as total_logs',
+            'SUM(log.is_scheduled) as pending_executions',
+            'SUM(CASE WHEN log.non_action_path_taken = 1 AND log.is_scheduled = 0 THEN 1 ELSE 0 END) as negative_path_count',
+            'SUM(CASE WHEN log.non_action_path_taken = 0 AND log.is_scheduled = 0 THEN 1 ELSE 0 END) as positive_path_count',
+            'MIN(log.date_triggered) as first_execution_date',
+            'MAX(log.date_triggered) as last_execution_date',
+        ])
+            ->from(MAUTIC_TABLE_PREFIX.'campaign_lead_event_log', 'log')
+            ->where(
+                $qb->expr()->and(
+                    $qb->expr()->eq('log.event_id', ':eventId')
+                )
+            )
+            ->setParameter('eventId', $eventId)
+            ->groupBy('log.event_id');
+
+        $result = $qb->executeQuery()->fetchAssociative();
+        if (false === $result) {
+            return [
+                'total_executions'     => 0,
+                'pending_executions'   => 0,
+                'negative_path_count'  => 0,
+                'positive_path_count'  => 0,
+                'first_execution_date' => null,
+                'last_execution_date'  => null,
+            ];
+        }
+        $totalLogs         = (int) $result['total_logs'];
+        $pendingExecutions = (int) $result['pending_executions'];
+
+        return [
+            'total_executions'     => $totalLogs - $pendingExecutions,
+            'pending_executions'   => (int) $result['pending_executions'],
+            'negative_path_count'  => (int) $result['negative_path_count'],
+            'positive_path_count'  => (int) $result['positive_path_count'],
+            'first_execution_date' => $result['first_execution_date'] ?: null,
+            'last_execution_date'  => $result['last_execution_date'] ?: null,
+        ];
```

Comment:
Do you think we could avoid the duplication like this?
```suggestion
        $totalLogs         = (int) ($result['total_logs'] ?? 0);
        $pendingExecutions = (int) ($result['pending_executions'] ?? 0);

        return [
            'total_executions'     => $totalLogs - $pendingExecutions,
            'pending_executions'   => (int) $result['pending_executions'] ?? 0,
            'negative_path_count'  => (int) $result['negative_path_count'] ?? 0,
            'positive_path_count'  => (int) $result['positive_path_count'] ?? 0,
            'first_execution_date' => $result['first_execution_date'] ?? null,
            'last_execution_date'  => $result['last_execution_date'] ?? null,
        ];
```

Reply:
When looking at how the array is used later I'd even suggest to create a DTO class for this data instead of associative array to avoid more checks like
```php
$lastExecutionDate  = $logStats['last_execution_date'] ? new DateTimeHelper($logStats['last_execution_date']) : null;
```